icu4j/main/classes/core/src/com/ibm/icu/impl/personname/PersonNameFormatterImpl.java

```diff
@@ -215,32 +224,84 @@ private String[] asStringArray(ICUResourceBundle resource) {
      * @param name The name to be formatted.
      * @return If true, use given-first order to format the name; if false, use surname-first order.
      */
-    private boolean nameIsGnFirst(PersonName name) {
-        // the name can declare its order-- check that first (it overrides any locale-based calculation)
-        if (name.getPreferredOrder() == PersonName.PreferredOrder.GIVEN_FIRST) {
-            return true;
-        } else if (name.getPreferredOrder() == PersonName.PreferredOrder.SURNAME_FIRST) {
-            return false;
+private boolean nameIsGnFirst(PersonName name) {
+    // the name can declare its order-- check that first (it overrides any locale-based calculation)
```

Comment:
Maybe put in some comments so we have the relationship to the spec?

```suggestion
// the name can declare its order-- check that first (it overrides any locale-based calculation)
// The numbered comments are from https://www.unicode.org/reports/tr35/tr35-personNames.html#derive-the-name-order
// 1. If the calling API requests sorting order, that is used.
// that is done outside of this method
```